elements/pl-order-blocks/pl-order-blocks.js

```diff
@@ -0,0 +1,120 @@
+            var answer_text = dom_objs[i].getAttribute('string');
+            var answer_indent = parseInt($(dom_objs[i]).css('marginLeft').replace('px', ''));
+            answer_indent = Math.round((answer_indent - INDENT_OFFSET) / TABWIDTH); // get how many times the answer is indented
+            student_answers.push(answer_text);
+            indents.push(answer_indent.toString());
+        }
+    }
+
+    textfield_name = event.target.getAttribute('name');
+    if (textfield_name === null){
+        return;
+    }
+    textfield_name = '#' + textfield_name + '-input';
+    answer_json.answers = student_answers;
+    answer_json.answer_indent = indents;
+    $(textfield_name).val(JSON.stringify(answer_json));
+}
+
+
+function update_indent(leftDiff, id, ui) {
+    if (!ui.item.parent()[0].classList.contains('dropzone') || !ui.item.parent()[0].classList.contains('enableIndentation')){
+        // no need to support indent on MCQ option panel or solution panel with indents explicitly disabled
+        ui.item[0].style.marginLeft = INDENT_OFFSET + 'px';
+        return;
+    }
+    leftDiff = ui.position.left - ui.item.parent().position().left;
+    var currentIndent = ui.item[0].style.marginLeft;
+    if (parseInt(currentIndent) <= MAX_INDENT + 1 && leftDiff < 0){
+        return; // if answer is not indented, and the student drag it left
+                // do nothing
+    }
+
+    leftDiff = (Math.round(leftDiff / TABWIDTH) * TABWIDTH);
+
+    // leftDiff is the direction to move the MCQ answer tile, in px
+    // we limit leftDiff to be increments of TABWIDTH, whether positive or negative
+    if (currentIndent != ''){
+        leftDiff += parseInt(currentIndent); 
+    }
+    // limit leftDiff to be in [INDENT_OFFSET, (TABWIDTH * MAX_INDENT) + INDENT_OFFSET], within the bounds of the drag and drop box
+    // that is, at least indented 0 times, or at most indented by MAX_INDENT times  
+    leftDiff = Math.min(Math.max(leftDiff, INDENT_OFFSET), (TABWIDTH * MAX_INDENT) + INDENT_OFFSET);
+
+    // when the user drag a tile into the answer box for the first time
+    // the snap to grid dragging doesnt apply
+    // so we have to manually enforce "snapping the leftDiff number to the nearest grid number" here
+    var remainder = (leftDiff - INDENT_OFFSET) % TABWIDTH;
+    if (remainder != 0) {
+        // Manually snap to grid here, by rounding to the nearest multiple of TABWIDTH
+        if (remainder > (TABWIDTH / 2)){
+            leftDiff += remainder; // round towards +∞, to the next bigger multiple of TABWIDTH
+        } else {
+            leftDiff -= remainder; // round towards -∞, to the next smaller multiple of TABWIDTH
+        }
+    }
+
+    ui.item[0].style.marginLeft = Math.max(leftDiff, INDENT_OFFSET) + 'px';
+}
+
+
+$( document ).ready(function() {
+    // Add drag and drop functionality for options elements
+    // that has the connectedSortable class
+    // code for the HTML popover
```

Comment:
Would look a little nicer with this on two longer lines instead of three short ones.

Reply:
fixed